core/src/main/scala/kafka/log/UnifiedLog.scala

```diff
@@ -136,16 +136,16 @@ class UnifiedLog(@volatile var logStartOffset: Long,
    */
   @volatile private var firstUnstableOffsetMetadata: Option[LogOffsetMetadata] = None
 
+  @volatile var partitionMetadataFile: Option[PartitionMetadataFile] = None
```

Comment:
nit: You can leave it at the earliest place for this field as it is not really needed for this change.